prow/cmd/checkconfig/main.go

```diff
@@ -197,6 +202,24 @@ func main() {
 			errs = append(errs, err)
 		}
 	}
+	if o.warningEnabled(unknownFieldsWarning) {
```

Comment:
Instead of implementing this using reflection, could we use the YAML library with the strict flag set and just emit the warning here?

Reply:
I agree that it would be nicer to not have the reflection. 

It is possible to use strict flag but it would only show one unknown field while this implementation would list all of them. While it is still beneficial I believe it could be misleading: if I run a tool and it shows that I have unknown field and it's only one I would expect that's the only unknown field in the config. And as was discussed it seems to be a legitimate case to have not-yet-implemented configs so I might think that config is valid except for one field that I know is not valid, which won't actually be the case.

I was trying to come up with a better approach, e.g. to serialize config to yaml and compare with original (won't work because ordering could be different), [asked on stackoverflow](https://stackoverflow.com/q/55029335/518469) but didn't find anything other than reflection.

Reply:
Example of multiple invalid properties: https://github.com/kubernetes/test-infra/pull/11622/files#diff-1bccbd8d60a651295f7c8bb39ac87d4dR463